[CI] GetPipelineRequestTests.testEqualsAndHashcode fails on master

**Build scan**:
https://gradle-enterprise.elastic.co/s/jr5nbkgcsbkxg

**Repro line**:
```
./gradlew ':x-pack:plugin:logstash:test' --tests "org.elasticsearch.xpack.logstash.action.GetPipelineRequestTests.testEqualsAndHashcode" -Dtests.seed=4228CC62F41F1E04 -Dtests.security.manager=true -Dtests.locale=en-AU -Dtests.timezone=Africa/Lubumbashi -Druntime.java=11
```

**Reproduces locally?**:
Yep

**Applicable branches**:
Master

**Failure history**:
I think related to the merging of #53350?

**Failure excerpt**:
```
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:logstash:test' --tests "org.elasticsearch.xpack.logstash.action.GetPipelineRequestTests.testEqualsAndHashcode" -Dtests.seed=4228CC62F41F1E04 -Dtests.security.manager=true -Dtests.locale=en-AU -Dtests.timezone=Africa/Lubumbashi -Druntime.java=11
  2> java.lang.AssertionError: GetPipelineRequest mutation should not be equal to original
    Expected: not <org.elasticsearch.xpack.logstash.action.GetPipelineRequest@20>
         but: was <org.elasticsearch.xpack.logstash.action.GetPipelineRequest@20>
        at __randomizedtesting.SeedInfo.seed([4228CC62F41F1E04:3327B4AF3BF8572B]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode(EqualsHashCodeTestUtils.java:82)
        at org.elasticsearch.test.AbstractWireTestCase.testEqualsAndHashcode(AbstractWireTestCase.java:60)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```

--- a/x-pack/plugin/logstash/src/test/java/org/elasticsearch/xpack/logstash/action/GetPipelineRequestTests.java
+++ b/x-pack/plugin/logstash/src/test/java/org/elasticsearch/xpack/logstash/action/GetPipelineRequestTests.java
@@ -28,10 +28,11 @@ public class GetPipelineRequestTests extends AbstractWireSerializingTestCase<Get
     @Override
     protected GetPipelineRequest mutateInstance(GetPipelineRequest instance) {
         List<String> ids = new ArrayList<>(instance.ids());
-        if (randomBoolean()) {
+        if (randomBoolean() || ids.size() == 0) {
             // append another ID
             ids.add(randomAlphaOfLengthBetween(2, 10));
         } else {
+            // change the strings in the request
             ids = ids.stream().map(id -> id + randomAlphaOfLength(1)).collect(Collectors.toList());
         }
         return new GetPipelineRequest(ids);

